functions/go/set-gcp-resource-ids/main.go

```diff
@@ -127,35 +110,51 @@ func (p *SetGCPProject) Transform(objects fn.KubeObjects) error {
 				displayName = displayName[:30]
 			}
 
-			object.SetNestedString(displayName, "spec", "name")     // name is the display name
-			object.SetNestedString(projectID, "spec", "resourceID") // resourceID is the project ID (must be unique)
+			// name is the display name
+			if err = object.SetNestedString(displayName, "spec", "name"); err != nil {
+				results.ErrorE(err)
+				return false
+			}
+			// resourceID is the project ID (must be unique)
+			if err = object.SetNestedString(projectID, "spec", "resourceID"); err != nil {
+				results.ErrorE(err)
+				return false
+			}
 		}
 
 		if object.GetAnnotation("cnrm.cloud.google.com/project-id") != "" {
-			object.SetAnnotation("cnrm.cloud.google.com/project-id", projectID)
+			if err = object.SetAnnotation( "cnrm.cloud.google.com/project-id", projectID); err != nil {
+				results.ErrorE(err)
+				return false
+			}
 		}
 
-		if object.IsGVK("core.cnrm.cloud.google.com", "v1beta1", "ConfigConnectorContext") {
+
+		if object.IsGroupVersionKind(ConfigControllerContext) {
 			// TODO: ConfigConnectorContext should accept a serviceAccountRef
 			googleServiceAccount, _, _ := object.NestedString("spec", "googleServiceAccount")
 			if googleServiceAccount != "" {
 				tokens := strings.Split(googleServiceAccount, "@")
 				if len(tokens) != 2 {
-					return fmt.Errorf("error parsing spec.googleServiceAccount=%q", googleServiceAccount)
+					results.Errorf("error parsing spec.googleServiceAccount=%q", googleServiceAccount)
+					return false
```

Comment:
I'm not entirely sure what the benefit is here, vs returning the error (?)

Reply:
return bool tells whether the function is considered "PASS" or "FAIL". in kpt, it would be
```
[PASS] "gcr.io/kpt-fn-demo/set-gcp-resource-ids" in 1.6s
```
or 
```
[FAIL] "gcr.io/kpt-fn-demo/set-gcp-resource-ids" in 1.6s
Error: exit status 1
```

The results can more debugging purpose, and each `fn.Result` can have different severity. 
```
  Results:
    [info]: namespace "example" updated to "example", 0 values changed
    [warning]: something unexpected..
```  
And what's more,  in a function pipeline, the results can be appended rather than replaced (I am not sure if this is what kpt supports today).

Reply:
OK, I would expect to put that into the result e.g. `result.MarkFailed("reason")`, but we can merge and iterate!